frontend/src/pages/Lessons/index.jsx

```diff
@@ -24,6 +30,13 @@ function Lessons() {
         .filter((lesson) => lesson.schoolTopic_id === parseInt(id, 10))
         .map((lesson) => (
           <LessonCard
+            isFav={fav
+              .filter((f) => {
+                return f.id === lesson.id;
+              })
+              .map(() => {
+                return true;
+              })}
```

Comment:
Uselessly complicated
Try something like the [find method](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/find):
```
idFav={ typeof fav.find(elt=>elt.id===lesson.id) === number }
```